compiler/rustc_monomorphize/src/collector.rs

```diff
@@ -765,7 +765,7 @@ impl<'a, 'tcx> MirVisitor<'tcx> for MirNeighborCollector<'a, 'tcx> {
                 ty::ConstKind::Unevaluated(ct) => {
```

Comment:
that should be unreachable again, all constants should already be evaluated

Reply:
also, generally I think  we should `bug!` if it's not a `ty::ConstKind::Value`

Reply:
also `collect_const_value`should be a noop for valtrees unless we allow functions.

So just completely ignore `mir::ConstantKind::Ty(_)` with a fixme
```rust
FIXME(adt_const_params): In case function pointers are allowed in valtrees,
we have to walk the valtree to collect all relevant functions.
```